pallets/allocations/src/lib.rs

```diff
@@ -83,6 +86,10 @@ pub mod pallet {
         #[pallet::weight(
 			<T as pallet::Config>::WeightInfo::allocate(proof.len() as u32)
 		)]
+        // we add the `transactional` modifier here in the event that one of the
+        // transfers fail. the code itself should already prevent this but we add
```

Comment:
If one transfer fails, this is a preventative measure to ensure the entire batch is rolled back. Correct?

Reply:
the batch would be rolled back if we use `batch_all` in the hot wallet (I believe we do?)

Reply:
To clarify, I was referring to the two `transfer` calls in the function in my comment. I was initially concerned one of them would fail silently and prevent the other one from executing correctly. Which would leave us with an inconsistent state unless I applied the `#[transactional]` modifier

Reply:
Indeed, the Hot Wallet handles transfers using `batch_all`. I verified this morning to be certain.